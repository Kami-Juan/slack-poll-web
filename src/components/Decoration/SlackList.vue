<template>
  <ul id="options-list">
    <slot />
  </ul>
</template>
<script>
export default {}
</script>
<style scoped>
#options-list {
  text-align: left;
  list-style-type: none;
  margin: 0;
  padding: 0;
}
</style>