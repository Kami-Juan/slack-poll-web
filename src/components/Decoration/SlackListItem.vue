<template>
  <li>
    <input type="text" class="option-input-element" v-model="option" />
    <a class="option-remove-element" @click="remove">‚ùå</a>
  </li>
</template>
<script>
export default {
  props: ['value'],
  methods: {
    remove() {
      this.$emit('remove')
    }
  },
  computed: {
    option: {
      get() {
        return this.value
      },
      set(event) {
        this.$emit('input', event)
      }
    }
  }
}
</script>
<style scoped>
.option-input-element {
  border: none;
}

.option-input-element:focus {
  outline: 0;
}

.option-remove-element {
  cursor: pointer;
}
</style>