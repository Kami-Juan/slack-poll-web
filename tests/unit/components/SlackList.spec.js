import { shallowMount } from '@vue/test-utils'
import SlackList from './../../../src/components/Decoration/SlackList'
import SlackListItem from './../../../src/components/Decoration/SlackListItem'

describe('test SlackList.vue', () => {
  let cmp

  beforeEach(() => {
    const slackListItemWrapper = {
      render(h) {
        return h(SlackListItem, { props: { value: 'One element' } })
      }
    }

    cmp = shallowMount(SlackList, {
      slots: {
        default: slackListItemWrapper
      }
    })
  })

  it('should be render', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  it('SlackListItem are inserted in the component', () => {
    const slackList = cmp.find(SlackList)
    expect(slackList.find(SlackListItem).isVueInstance()).toBe(true)
  })
})