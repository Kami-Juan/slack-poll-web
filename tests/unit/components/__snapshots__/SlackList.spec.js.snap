// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SlackList.vue should be render 1`] = `
<ul
  id="options-list"
>
  <anonymous-stub
    value="One element"
  />
</ul>
`;
